### PR TITLE
Update Date Format

### PR DESCRIPTION
**Pull request by leobcgov**
# Fixes or Implements [AB#10205](https://qslvic.visualstudio.com/304a1f8c-dace-4f85-adf3-bf563d5b3a39/_workitems/edit/10205)

* [X] Enhancement
* [ ] Bug
* [ ] Documentation

## Description
- change all date format to yyyy-MMM-dd
- update function tests.

If you are reviewing this PR, please adhere to the guidelines as [documented](https://github.com/bcgov/healthgateway/wiki/prguidance).

## Testing

* [ ] Unit Tests Updated
* [X] Functional Tests Updated
* [ ] Not Required

### Steps to Reproduce

If this is a bug, please provide details on how to reproduce the code.

### UI Changes

YES

### Browsers Tested

* [X] Chrome
* [ ] Safari
* [ ] Edge
* [ ] Firefox

Provide an explanation for any test(s) not completed.

## Notes/Additional Comments

Any additional notes or comments that may be relevant.  Include any specialized deployment steps here.

### CODE REVIEW

**Review by tiagograf-AOT — APPROVED** [review]

**Review by FuriousLlama — APPROVED** [review]

File: Apps/WebClient/src/ClientApp/src/components/modal/immunizationCard.vue [hunk 1]
```diff
@@ -71,7 +71,7 @@ export default class ImmunizationCardComponent extends Vue {
                 covidImmunizations[index].immunization.immunizationAgents[0];
             this.doses.push({
                 product: agent.productName,
-                date: this.formatDate(element.dateOfImmunization),
+                date: DateWrapper.format(element.dateOfImmunization),
```
Comment (MikeLyttle) [inline Apps/WebClient/src/ClientApp/src/components/modal/immunizationCard.vue:74]: Should this be `date: new DateWrapper(element.dateOfImmunization).format()`?

File: Apps/WebClient/src/ClientApp/src/components/datePicker.vue [hunk 2]
```diff
@@ -20,10 +19,11 @@ export default class DatePickerComponent extends Vue {
     @Prop() state?: boolean;
     @Ref("datePicker") datePicker!: BFormDatepicker;
```
Comment (FuriousLlama) [inline Apps/WebClient/src/ClientApp/src/components/datePicker.vue:20]: Test comment

File: Testing/functional/e2e/cypress/integration/report/report.js [hunk 3]
```diff
@@ -21,36 +21,37 @@ describe('Reports', () => {
...
+            .type("2020-FEB-05")
         cy.get('[data-testid=clearBtn]')
             .should('be.enabled', 'be.visible')
             .click();
         cy.get('[data-testid=startDateInput] input')
-            .should('have.value','2021-02-03')
+            .should('have.value','2021-FEB-03')
         cy.get('[data-testid=endDateInput] input')
-            .should('have.value','2021-02-05')
+            .should('have.value','2021-FEB-05')
```
Comment (MikeLyttle) [inline Testing/functional/e2e/cypress/integration/report/report.js:54]: For consistency, should the month be in title case ("Feb")?

File: Testing/functional/e2e/cypress/integration/timeline/filter.js [hunk 4]
```diff
@@ -63,11 +63,11 @@ describe("Filters", () => {
         cy.get("[data-testid=filterStartDateInput] input")
             .focus()
             .clear()
-            .type("2020-06-14");
+            .type("2020-JUN-14");
         cy.get("[data-testid=filterEndDateInput] input")
             .focus()
             .clear()
-            .type("2020-06-14");
+            .type("2020-JUN-14").focus();
```
Comment (MikeLyttle) [inline Testing/functional/e2e/cypress/integration/timeline/filter.js:70]: For consistency, should the month be in title case ("Oct" and "Jun")?

**Review by MikeLyttle — CHANGES_REQUESTED** [review]

File: Apps/WebClient/src/ClientApp/src/components/modal/immunizationCard.vue (hunk 1 above)
Comment (MikeLyttle) [reply Apps/WebClient/src/ClientApp/src/components/modal/immunizationCard.vue:74]: I might be wrong on this one - I just found the static method.

File: Testing/functional/e2e/cypress/integration/timeline/filter.js (hunk 4 above)
Comment (MikeLyttle) [reply Testing/functional/e2e/cypress/integration/timeline/filter.js:70]: Leo explained that we have a mask in place that capitalizes these particular inputs.

**Review by MikeLyttle — APPROVED** [review]

**Pull request merged**